hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/listener/ClientSmartListenerService.java

```diff
@@ -143,22 +138,13 @@ public void shutdown() {
 
     @Override
     public void memberAdded(final MembershipEvent membershipEvent) {
-        executionService.executeInternal(new Runnable() {
-            @Override
-            public void run() {
-                synchronized (listenerRegLock) {
-                    Member member = membershipEvent.getMember();
-                    members.add(member);
-                    for (ClientRegistrationKey registrationKey : registrations.keySet()) {
-                        try {
-                            invoke(registrationKey, member.getAddress());
-                        } catch (Exception e) {
-                            logger.warning("Listener " + registrationKey + " can not added to new member " + member);
-                        }
-                    }
-                }
+        synchronized (listenerRegLock) {
```

Comment:
This is not a new logic. When I was first implementing this one, I start with a single thread. It seemed more complicated to reviewers back then, and I switched to locks.  In short, lock is not related to this pr.